pkg/operator/ceph/cluster/osd/labels.go

```diff
@@ -60,6 +60,7 @@ func (c *Cluster) getOSDLabels(osd OSDInfo, failureDomainValue string, portable
 	labels[portableKey] = strconv.FormatBool(portable)
 	labels[deviceClass] = osd.DeviceClass
```

Comment:
Will upgraded OSDs get the correct device class properties set? Since the osd prepare job is skipped during upgrades for pvc-based clusters, the invalid device class name may still be a bug.

Reply:
If the osd-prepare job didn't run, it would create/update the communication cm for osd's, and the same information would be present in the OSDInfo, which would not change the label value,

@sp98 @BlaineEXE do you have any idea?

Reply:
@travisn I tried on a lso based cluster with PVC,

The updating the image doesn't worked to update the label, but once I deleted the osd deployment it did changed the device class label, because internal rook-operator re-stated the prepare pod job,


So do we delete the odds deployments with upgrades or if not how we wanna tackle this scenario?

Reply:
OPened a tracker for upgrades https://github.com/rook/rook/issues/15082

Reply:
Agreed, let's not re-trigger the OSD prepare. At least this PR will fix the labels for new OSDs.